flaskr/follows.py

```diff
@@ -72,5 +77,66 @@ def get_followed(username):
         .join(Follow, Follow.followed_id == User.id) \
         .filter(Follow.follower_id == target_id) \
         .all()
-    print(followed)
     return followed
+
+
+# User recommandation
+class FollowGraph:
+
+    def __init__(self):
+        self.edges = dict()  # from vertices to vertices
+
+    def delete_user(self, user_id: int):
+        try:
+            del self.edges[user_id]
+            for key in self.edges:
+                self.edges[key].remove(user_id)
+        except KeyError:
+            print("Unknown user")
+
+    def add_follow(self, follower_id: int, followed_id: int):
+        try:
+            self.edges[follower_id].add(followed_id)
+        except KeyError:
+            self.edges[follower_id] = {followed_id}
+
+    def remove_follow(self, follower_id: int, followed_id: int):
+        try:
+            self.edges[follower_id].remove(followed_id)
+            if len(self.edges[follower_id]) == 0:
+                del self.edges[follower_id]
+        except KeyError:
+            print("Unknown user")
+
+    def recommandation(self, source_user: User, max_depth: int):
+        if source_user.id not in self.edges:
+            print("not following anyone, random suggestions")
+            return random.sample(self.edges, min(max_depth, len(self.edges)))
+        result = set()
+        explored_set = set()
+        frontier = [source_user.id]
+        current_depth = 0
+        while len(frontier) > 0 and current_depth <= max_depth:
+            current_id = frontier[0]
+            explored_set.add(current_id)
+            for neighboor_id in self.edges[current_id]:
+                frontier.append(neighboor_id)
+                if neighboor_id not in self.edges[source_user.id]:
+                    result.add(neighboor_id)
+            current_depth += 1
+        return result
+
+    def print(self):
+        for key in self.edges:
+            print("User {} follows {}".format(key, self.edges[key]))
+
+
+# Global follow variable
+follow_graph = FollowGraph()  # need to init it with init_follow_graph
```

Comment:
any way we can avoid this global var? Would be great if we could place it in the `session` object or something like that